src/Core/Framework/App/Subscriber/SystemLanguageChangedSubscriber.php

```diff
@@ -0,0 +1,113 @@
+    {
+        return [
+            SystemLanguageChangeEvent::class => 'onSystemLanguageChanged',
+        ];
+    }
+
+    public function onSystemLanguageChanged(SystemLanguageChangeEvent $event): void
+    {
+        // This repository is part of the Administration bundle and not necessarily installed and available.
+        if ($this->snippetRepository === null) {
+            return;
+        }
+
+        $context = Context::createDefaultContext();
+
+        $snippets = $this->getSnippets($this->snippetRepository, $context);
+        if ($snippets->count() === 0) {
+            return;
+        }
+
+        $appIds = $this->getApps($context)->getIds();
+
+        $appsWithSnippets = [];
+        foreach ($snippets as $snippet) {
+            if (\in_array($snippet->getAppId(), $appIds, true) && !\in_array($snippet->getAppId(), $appsWithSnippets, true)) {
+                $appsWithSnippets[] = $snippet->getAppId();
+            }
+        }
+
+        $previousLocale = $this->getLocale($event->previousLanguageId, $context);
+        $newLocale = $this->getLocale(Defaults::LANGUAGE_SYSTEM, $context);
+
+        foreach ($appsWithSnippets as $appId) {
+            $snippetToClone = $snippets->filter(fn (AppAdministrationSnippetEntity $snippet) => $appId === $snippet->getAppId() && $snippet->getLocaleId() === $newLocale->getId())->first();
+            \assert($snippetToClone instanceof AppAdministrationSnippetEntity);
+
+            $snippetWithPreviousLocaleExists = $snippets->filter(fn (AppAdministrationSnippetEntity $snippet) => $appId === $snippet->getAppId() && $snippet->getLocaleId() === $previousLocale->getId())->first();
+            if ($snippetWithPreviousLocaleExists) {
+                continue;
+            }
+
+            $this->snippetRepository->create([[
+                'appId' => $snippetToClone->getAppId(),
+                'localeId' => $previousLocale->getId(),
+                'value' => $snippetToClone->getValue(),
+            ]], $context);
+        }
+    }
+
+    private function getLocale(string $languageId, Context $context): LocaleEntity
+    {
+        $criteria = new Criteria([$languageId]);
+        $criteria->addAssociation('locale');
+
+        $language = $this->languageRepository->search($criteria, $context)->first();
+        \assert($language instanceof LanguageEntity);
+
+        $locale = $language->getLocale();
+        \assert($locale instanceof LocaleEntity);
+
+        return $locale;
+    }
+
+    private function getApps(Context $context): AppCollection
+    {
+        return $this->appRepository->search(new Criteria(), $context)->getEntities();
+    }
+
+    /**
+     * @param EntityRepository<AppAdministrationSnippetCollection> $snippetRepository
+     */
+    private function getSnippets(EntityRepository $snippetRepository, Context $context): AppAdministrationSnippetCollection
+    {
+        return $snippetRepository->search(new Criteria(), $context)->getEntities();
+    }
```

Comment:
The repo is already available:

```suggestion
    private function getSnippets(Context $context): AppAdministrationSnippetCollection
    {
        return $this->snippetRepository->search(new Criteria(), $context)->getEntities();
    }
```

Reply:
It's nullable in the constructor and this way I make PHPStan happy without having to do additional `null` checks 😂